### ISSUE TITLE
Filter By Value

### ISSUE THREAD

**Issue by KashifAhmed** [angular/angularfire2#133]
I can't find that how can I filter the list with angularFire2, there is no method .toEqual or something like that.

**Comment by davideast** [comment]
We are working on a declarative querying model, see #16.

For now you can query on a ref and pass that into a list or object.

```ts
constructor(af: AngularFire, @Inject(FirebaseRef) ref: FirebaseRef) {
  const query = ref.child('items').orderByChild('order').equalTo('1');
  const list = af.database.list(query);
}
```

**Comment by mip1983** [comment]
@davideast Hi David, just following the example there. My IDE/typescript compiler (visual studio) doesn't like that code as the type I'm passing comes up as 'FirebaseQuery' rather than 'Firebase'.

Maybe the list function needs to be tweaked to:

list(urlOrRef: string | Firebase | FirebaseQuery...

**Comment by joaogarin** [comment]
Hello,

I was trying something similar to this and get the same thing as @mip1983 is mentioning.

"Argument of type 'FirebaseQuery' is not assignable to parameter of type 'string'"

**Comment by joaogarin** [comment]
This was merged in alpha 16 just noticed ;) cool

**Comment by mip1983** [comment]
@joaogarin Not sure it's merged in there yet: https://github.com/angular/angularfire2/pull/138

**Comment by joaogarin** [comment]
Yeah I just noticed was going through the issues tks for the info !;)

**Comment by Seaborg** [comment]
Is this already implemented in 2.0.0-alpha.16 or not?

**Comment by joaogarin** [comment]
No this not yet implemented. You can follow it here https://github.com/angular/angularfire2/pull/138
